sources/logdog/src/main.rs

```diff
@@ -0,0 +1,256 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+/*!
+# Introduction
+
+`logdog` is a program that gathers logs from various places on a Bottlerocket host and combines them
+into a tarball for easy export.
+
+Usage example:
+```
+$ logdog
+logs are at: /tmp/bottlerocket-logs.tar.gz
+```
+
+### Logs Collected
+
+`logdog` will aggregate the following information:
+
+ * a copy of os-release to tell us the version and build of Bottlerocket
+ * a list of boots that journalctl knows about
+ * errors from journalctl
+ * all log lines from journalctl
+ * signpost status to tell us the status of grub and the boot partitions
+ * Bottlerocket settings using the apiclient
+ * networking status from wicked
+ * configuration info from containerd
+ * the status of kubelet and other kube processes from systemctl
+ * the kernel message buffer with dmesg
+ * firewall filtering information from iptables
+ * firewall nat information from iptables
+ * disk and filesytem information from df
+
+*/
+
+#![deny(rust_2018_idioms)]
+
+mod create_tarball;
+mod error;
+mod exec_to_file;
+mod temp_dir;
+
+use std::fs::remove_dir_all;
+use std::path::{Path, PathBuf};
+use std::{env, process};
+
+use create_tarball::create_tarball;
+use error::{FileError, IoError, Result};
+use exec_to_file::{run_commands, ExecToFile};
+use temp_dir::TempDir;
+
+use snafu::{ErrorCompat, ResultExt};
+
+const ERROR_FILENAME: &str = "logdog.errors";
+const OUTPUT_FILENAME: &str = "bottlerocket-logs.tar.gz";
+const TARBALL_DIRNAME: &str = "bottlerocket-logs";
+const TEMPDIR_NAME: &str = "logdog-temp";
+
+/// Prints a usage message in the event a bad arg is passed.
+fn usage() -> ! {
+    let program_name = env::args().next().unwrap_or_else(|| "program".to_string());
+    eprintln!(
+        r"Usage: {}
+            [ --output PATH file to write zipped logs to ]
+",
+        program_name,
+    );
+    process::exit(2);
+}
+
+/// Prints a more specific message before exiting through usage().
+fn usage_msg(msg: &str) -> ! {
+    eprintln!("{}\n", msg);
+    usage();
+}
+
+/// Parses the command line arguments.
+fn parse_args(args: env::Args) -> PathBuf {
+    let mut output_arg = None;
+    let mut iter = args.skip(1);
+    while let Some(arg) = iter.next() {
+        match arg.as_ref() {
+            "--output" => {
+                output_arg = Some(
+                    iter.next()
+                        .unwrap_or_else(|| usage_msg("Did not give argument to --output")),
+                )
+            }
+
+            _ => usage(),
+        }
+    }
+
+    match output_arg {
+        Some(path) => PathBuf::from(path),
+        None => env::temp_dir().as_path().join(OUTPUT_FILENAME),
+    }
+}
+
+/// Runs the bulk of the program's logic, main wraps this.
+fn run_program(output: &PathBuf) -> Result<()> {
```

Comment:
```suggestion
fn run(output: &PathBuf) -> Result<()> {
```

To match convention.